packages/process/src/node/raw-process.ts

```diff
@@ -19,85 +19,73 @@ import { ProcessManager } from './process-manager';
+
+            // About catching errors: spawn will sometimes throw directly
+            // (EACCES on Linux), sometimes return a Process object with the pid
+            // property undefined (ENOENT on Linux).  In the latter case, wait
+            // for the error event to reject the Promise.
+            const process: ChildProcess = spawn(options.command, options.args, options.options);
+            if (process.pid === undefined) {
+                process.on('error', (err: Error) => {
+                    reject(err);
+                });
+            } else {
+                const rawProcess = new RawProcess(process, this.logger);
+                this.processManager.register(rawProcess);
+                resolve(rawProcess);
+            }
+        });
     }
 }
 
-@injectable()
 export class RawProcess extends Process {
 
-    readonly input: stream.Writable;
-    readonly output: stream.Readable;
-    readonly errorOutput: stream.Readable;
-    readonly process: ChildProcess;
-
-    constructor(
-        @inject(RawProcessOptions) options: RawProcessOptions,
-        @inject(ProcessManager) processManager: ProcessManager,
-        @inject(ILogger) @named('process') logger: ILogger
-    ) {
-        super(processManager, logger, ProcessType.Raw, options);
-
-        this.logger.debug(`Starting raw process: ${options.command},`
-            + ` with args: ${options.args ? options.args.join(' ') : ''}, `
-            + ` with options: ${JSON.stringify(options.options)}`);
-
-        /* spawn can throw exceptions, for example if the file is not
-           executable, it throws an error with EACCES.  Here, we try to
-           normalize the error handling by calling the error handler
-           instead.  */
-        try {
-            this.process = spawn(
-                options.command,
-                options.args,
-                options.options);
-
-            this.process.on('error', this.emitOnError.bind(this));
-            this.process.on('exit', this.emitOnExit.bind(this));
-
-            this.output = this.process.stdout;
-            this.input = this.process.stdin;
-            this.errorOutput = this.process.stderr;
-        } catch (error) {
-            /* When an error is thrown, set up some fake streams, so the client
-               code doesn't break because these field are undefined.  */
-            this.output = new DevNullStream();
-            this.input = new DevNullStream();
-            this.errorOutput = new DevNullStream();
-
-            /* Call the client error handler, but first give them a chance to register it.  */
-            process.nextTick(() => {
-                this.errorEmitter.fire(error);
-            });
-        }
+    constructor(readonly process: ChildProcess, logger: ILogger) {
+        super(logger, ProcessType.Raw);
+
+        this.process.on('exit', this.emitOnExit.bind(this));
     }
 
     get pid() {
         return this.process.pid;
     }
 
+    get stdin() {
```

Comment:
I have renamed input to stdin, output to stdout, errorOutput to stderr. It is not a necessary change, so I can revert that to reduce the number of breaking changes.